apis/hive/v1/vsphere/platform.go

```diff
@@ -27,6 +27,10 @@ type Platform struct {
 	// virtual machines.
 	Folder string `json:"folder,omitempty"`
 
+	// ResourcePool is the name of the resource pool that will be used and/or created
+	// for virtual machines.
+	ResourcePool string `json:"resourcePool,omitempty"`
```

Comment:
@jcpowermac points out that the upstream version of this field is `+optional`. That may not be significant since it's a non-pointer type, but might as well make it match.

Sounds like the comment could use a tweak too: nix `and/or created` since installer doesn't (can't) create RPs.

Reply:
Also as [discussed](https://redhat-internal.slack.com/archives/CE3ETN3J8/p1696881945939289), we want the new field to be in MachinePool, not ClusterDeployment. Day 0, the user (optionally) specifies a RP via install-config; day 2 they (optionally) specify a (possibly different) RP via MachinePool.